src/Algebra/Module/Properties/Semimodule.agda

```diff
@@ -0,0 +1,40 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of semimodules.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                using (CommutativeSemiring)
+open import Algebra.Module.Bundles using (Semimodule)
+open import Level                  using (Level)
+
+module Algebra.Module.Properties.Semimodule
+  {r ℓr m ℓm : Level}
+  {semiring  : CommutativeSemiring r ℓr}
+  (semimod   : Semimodule semiring m ℓm)
+  where
+
+open CommutativeSemiring semiring
+open Semimodule          semimod
+open import Relation.Nullary.Negation using (contraposition)
+open import Relation.Binary.Reasoning.Setoid ≈ᴹ-setoid
+
+s≈0⇒s*v≈0 : ∀ {s} {v} → s ≈ 0# → s *ₗ v ≈ᴹ 0ᴹ
```

Comment:
This is true for any left zero (and the next, for any right zero). I get the feeling that it might already be proved in greater generality elsewhere? (Or should be?)

Reply:
Hi @JacquesCarette ,

I'm guessing that you'd like to see these properties moved into `Algebra.Module.Structures.IsLeftSemimodule`.
Is that correct?
It makes sense, since that's where the last dependencies (i.e. - `*ₗ-congˡ` and `*ₗ-congʳ`) are defined.

However, when I try to move this code into that record definition, I bump into all kinds of awkwardness, which I believe stems from the fact that `*ₗ` is _not_ defined as an infix operator:

```agda
record IsLeftSemimodule (*ₗ : Opₗ R M) : Set (r ⊔ m ⊔ ℓr ⊔ ℓm) where
```

Is this movement of code super important to you?

Reply:
But, doesn't it need to apply, at least, to a semimodule?
That is, isn't _left zero_ a (semi)module-centric conept?